lib/types/src/compilation/function.rs

```diff
@@ -45,6 +48,42 @@ pub struct FunctionBody {
     pub unwind_info: Option<CompiledFunctionUnwindInfo>,
 }
 
+/// Any struct that acts like a `FunctionBody`.
+#[allow(missing_docs)]
+pub trait FunctionBodyLike<'a> {
```

Comment:
I'm not a fan of the `*Like` naming scheme.

But then again, I don't have a significantly better idea...

The best I can come up with is `*Accessor` or `*Reader`, but I don't think those are great either.

We can leave it as is if we don't have other ideas.

Reply:
I'm not a fan of *Like, *Accessor or *Reader either. Open to suggestions though!